File: ui/src/domain/access.ts
export interface AccessModel extends BaseModel {
  name: string;
  provider: string;
  config: /*
    注意：如果追加新的类型，请保持以 ASCII 排序。
    NOTICE: If you add new type, please keep ASCII order.
  */ Record<string, unknown> &
    (
      | AccessConfigFor1Panel
      | AccessConfigForACMECA
      | AccessConfigForACMEHttpReq
      | AccessConfigForAliyun
      | AccessConfigForAPISIX
      | AccessConfigForAWS
      | AccessConfigForAzure
      | AccessConfigForBaiduCloud
      | AccessConfigForBaishan
      | AccessConfigForBaotaPanel
      | AccessConfigForBaotaWAF
      | AccessConfigForBunny
      | AccessConfigForBytePlus
      | AccessConfigForCacheFly
      | AccessConfigForCdnfly
      | AccessConfigForCloudflare
      | AccessConfigForClouDNS
      | AccessConfigForCMCCCloud
      | AccessConfigForConstellix
      | AccessConfigForCTCCCloud
      | AccessConfigForDeSEC
      | AccessConfigForDigitalOcean
      | AccessConfigForDingTalkBot
      | AccessConfigForDiscordBot
      | AccessConfigForDNSLA
      | AccessConfigForDogeCloud
      | AccessConfigForDuckDNS
      | AccessConfigForDynv6
      | AccessConfigForEdgio
      | AccessConfigForEmail
      | AccessConfigForFlexCDN
      | AccessConfigForGcore
      | AccessConfigForGname
      | AccessConfigForGoDaddy
      | AccessConfigForGoEdge
      | AccessConfigForGoogleTrustServices
      | AccessConfigForHetzner
      | AccessConfigForHuaweiCloud
      | AccessConfigForJDCloud
      | AccessConfigForKubernetes
      | AccessConfigForLarkBot
      | AccessConfigForLeCDN
      | AccessConfigForMattermost
      | AccessConfigForNamecheap
      | AccessConfigForNameDotCom
      | AccessConfigForNameSilo
      | AccessConfigForNetcup
      | AccessConfigForNetlify
      | AccessConfigForPorkbun
      | AccessConfigForPowerDNS
      | AccessConfigForProxmoxVE
      | AccessConfigForQiniu
      | AccessConfigForRainYun
      | AccessConfigForRatPanel
      | AccessConfigForSafeLine
      | AccessConfigForSlackBot
      | AccessConfigForSSH
      | AccessConfigForSSLCom
      | AccessConfigForTelegramBot
      | AccessConfigForTencentCloud
      | AccessConfigForUCloud
      | AccessConfigForUniCloud
      | AccessConfigForUpyun
      | AccessConfigForVercel
      | AccessConfigForVolcEngine
      | AccessConfigForWangsu
      | AccessConfigForWebhook
      | AccessConfigForWeComBot
      | AccessConfigForWestcn
      | AccessConfigForZeroSSL
    );
  reserve?: "ca" | "notification";
}

// #region AccessConfig
export type AccessConfigFor1Panel = {
  serverUrl: string;
  apiVersion: string;
  apiKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForACMECA = {
  endpoint: string;
  eabKid?: string;
  eabHmacKey?: string;
};

export type AccessConfigForACMEHttpReq = {
  endpoint: string;
  mode?: string;
  username?: string;
  password?: string;
};

export type AccessConfigForAliyun = {
  accessKeyId: string;
  accessKeySecret: string;
  resourceGroupId?: string;
};

export type AccessConfigForAPISIX = {
  serverUrl: string;
  apiKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForAWS = {
  accessKeyId: string;
  secretAccessKey: string;
};

export type AccessConfigForAzure = {
  tenantId: string;
  clientId: string;
  clientSecret: string;
  environment?: string;
};

export type AccessConfigForBaiduCloud = {
  accessKeyId: string;
  secretAccessKey: string;
};

export type AccessConfigForBaishan = {
  apiToken: string;
};

export type AccessConfigForBaotaPanel = {
  serverUrl: string;
  apiKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForBaotaWAF = {
  serverUrl: string;
  apiKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForBunny = {
  apiKey: string;
};

export type AccessConfigForBytePlus = {
  accessKey: string;
  secretKey: string;
};

export type AccessConfigForCacheFly = {
  apiToken: string;
};

export type AccessConfigForCdnfly = {
  serverUrl: string;
  apiKey: string;
  apiSecret: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForCloudflare = {
  dnsApiToken: string;
  zoneApiToken?: string;
};

export type AccessConfigForClouDNS = {
  authId: string;
  authPassword: string;
};

export type AccessConfigForCMCCCloud = {
  accessKeyId: string;
  accessKeySecret: string;
};

export type AccessConfigForConstellix = {
  apiKey: string;
  secretKey: string;
};

export type AccessConfigForCTCCCloud = {
  accessKeyId: string;
  secretAccessKey: string;
};

export type AccessConfigForDeSEC = {
  token: string;
};

export type AccessConfigForDigitalOcean = {
  accessToken: string;
};

export type AccessConfigForDingTalkBot = {
  webhookUrl: string;
  secret?: string;
};

export type AccessConfigForDiscordBot = {
  botToken: string;
  defaultChannelId?: string;
};

export type AccessConfigForDNSLA = {
  apiId: string;
  apiSecret: string;
};

export type AccessConfigForDogeCloud = {
  accessKey: string;
  secretKey: string;
};

export type AccessConfigForDuckDNS = {
  token: string;
};

export type AccessConfigForDynv6 = {
  httpToken: string;
};

export type AccessConfigForEdgio = {
  clientId: string;
  clientSecret: string;
};

export type AccessConfigForEmail = {
  smtpHost: string;
  smtpPort: number;
  smtpTls: boolean;
  username: string;
  password: string;
  defaultSenderAddress?: string;
  defaultReceiverAddress?: string;
};

export type AccessConfigForFlexCDN = {
  serverUrl: string;
  apiRole: string;
  accessKeyId: string;
  accessKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForGcore = {
  apiToken: string;
};

export type AccessConfigForGname = {
  appId: string;
  appKey: string;
};

export type AccessConfigForGoDaddy = {
  apiKey: string;
  apiSecret: string;
};

export type AccessConfigForGoEdge = {
  serverUrl: string;
  apiRole: string;
  accessKeyId: string;
  accessKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForGoogleTrustServices = {
  eabKid: string;
  eabHmacKey: string;
};

export type AccessConfigForHetzner = {
  apiToken: string;
};

export type AccessConfigForHuaweiCloud = {
  accessKeyId: string;
  secretAccessKey: string;
  enterpriseProjectId?: string;
};

export type AccessConfigForJDCloud = {
  accessKeyId: string;
  accessKeySecret: string;
};

export type AccessConfigForKubernetes = {
  kubeConfig?: string;
};

export type AccessConfigForLarkBot = {
  webhookUrl: string;
};

export type AccessConfigForLeCDN = {
  serverUrl: string;
  apiVersion: string;
  apiRole: string;
  username: string;
  password: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForMattermost = {
  serverUrl: string;
  username: string;
  password: string;
  defaultChannelId?: string;
};

export type AccessConfigForNamecheap = {
  username: string;
  apiKey: string;
};

export type AccessConfigForNameDotCom = {
  username: string;
  apiToken: string;
};

export type AccessConfigForNameSilo = {
  apiKey: string;
};

export type AccessConfigForNetcup = {
  customerNumber: string;
  apiKey: string;
  apiPassword: string;
};

export type AccessConfigForNetlify = {
  apiToken: string;
};

export type AccessConfigForNS1 = {
  apiKey: string;
};

export type AccessConfigForPorkbun = {
  apiKey: string;
  secretApiKey: string;
};

export type AccessConfigForPowerDNS = {
  serverUrl: string;
  apiKey: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForProxmoxVE = {
  serverUrl: string;
  apiToken: string;
  apiTokenSecret?: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForQiniu = {
  accessKey: string;
  secretKey: string;
};

export type AccessConfigForRainYun = {
  apiKey: string;
};

export type AccessConfigForRatPanel = {
  serverUrl: string;
  accessTokenId: number;
  accessToken: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForSafeLine = {
  serverUrl: string;
  apiToken: string;
  allowInsecureConnections?: boolean;
};

export type AccessConfigForSlackBot = {
  botToken: string;
  defaultChannelId?: string;
};

export type AccessConfigForSSH = {
  host: string;
  port: number;
  authMethod?: string;
  username?: string;
  password?: string;
  key?: string;
  keyPassphrase?: string;
};

export type AccessConfigForSSLCom = {
  eabKid: string;
  eabHmacKey: string;
};

export type AccessConfigForTelegramBot = {
  botToken: string;
  defaultChatId?: number;
};

export type AccessConfigForTencentCloud = {
  secretId: string;
  secretKey: string;
};

export type AccessConfigForUCloud = {
  privateKey: string;
  publicKey: string;
  projectId?: string;
};

export type AccessConfigForUniCloud = {
  username: string;
  password: string;
};

export type AccessConfigForUpyun = {
  username: string;
  password: string;
};

export type AccessConfigForVercel = {
  apiAccessToken: string;
  teamId?: string;
};

export type AccessConfigForVolcEngine = {
  accessKeyId: string;
  secretAccessKey: string;
};

export type AccessConfigForWangsu = {
  accessKeyId: string;
  accessKeySecret: string;
  apiKey: string;
};

export type AccessConfigForWebhook = {
  url: string;
  method: string;
  headers?: string;
  allowInsecureConnections?: boolean;
  defaultDataForDeployment?: string;
  defaultDataForNotification?: string;
};

export type AccessConfigForWeComBot = {
  webhookUrl: string;
};

export type AccessConfigForWestcn = {
  username: string;
  apiPassword: string;
};

export type AccessConfigForZeroSSL = {
  eabKid: string;
  eabHmacKey: string;
};
// #endregion

File: internal/domain/access.go
package domain

import (
	"time"
)

const CollectionNameAccess = "access"

type Access struct {
	Meta
	Name      string         `json:"name" db:"name"`
	Provider  string         `json:"provider" db:"provider"`
	Config    map[string]any `json:"config" db:"config"`
	Reserve   string         `json:"reserve,omitempty" db:"reserve"`
	DeletedAt *time.Time     `json:"deleted" db:"deleted"`
}

type AccessConfigFor1Panel struct {
	ServerUrl                string `json:"serverUrl"`
	ApiVersion               string `json:"apiVersion"`
	ApiKey                   string `json:"apiKey"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForACMECA struct {
	Endpoint   string `json:"endpoint"`
	EabKid     string `json:"eabKid,omitempty"`
	EabHmacKey string `json:"eabHmacKey,omitempty"`
}

type AccessConfigForACMEHttpReq struct {
	Endpoint string `json:"endpoint"`
	Mode     string `json:"mode,omitempty"`
	Username string `json:"username,omitempty"`
	Password string `json:"password,omitempty"`
}

type AccessConfigForAliyun struct {
	AccessKeyId     string `json:"accessKeyId"`
	AccessKeySecret string `json:"accessKeySecret"`
}

type AccessConfigForAWS struct {
	AccessKeyId     string `json:"accessKeyId"`
	SecretAccessKey string `json:"secretAccessKey"`
}

type AccessConfigForAzure struct {
	TenantId     string `json:"tenantId"`
	ClientId     string `json:"clientId"`
	ClientSecret string `json:"clientSecret"`
	CloudName    string `json:"cloudName,omitempty"`
}

type AccessConfigForBaiduCloud struct {
	AccessKeyId     string `json:"accessKeyId"`
	SecretAccessKey string `json:"secretAccessKey"`
}

type AccessConfigForBaishan struct {
	ApiToken string `json:"apiToken"`
}

type AccessConfigForBaotaPanel struct {
	ServerUrl                string `json:"serverUrl"`
	ApiKey                   string `json:"apiKey"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForBaotaWAF struct {
	ServerUrl                string `json:"serverUrl"`
	ApiKey                   string `json:"apiKey"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForBytePlus struct {
	AccessKey string `json:"accessKey"`
	SecretKey string `json:"secretKey"`
}

type AccessConfigForBunny struct {
	ApiKey string `json:"apiKey"`
}

type AccessConfigForCacheFly struct {
	ApiToken string `json:"apiToken"`
}

type AccessConfigForCdnfly struct {
	ServerUrl                string `json:"serverUrl"`
	ApiKey                   string `json:"apiKey"`
	ApiSecret                string `json:"apiSecret"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForCloudflare struct {
	DnsApiToken  string `json:"dnsApiToken"`
	ZoneApiToken string `json:"zoneApiToken,omitempty"`
}

type AccessConfigForClouDNS struct {
	AuthId       string `json:"authId"`
	AuthPassword string `json:"authPassword"`
}

type AccessConfigForCMCCCloud struct {
	AccessKeyId     string `json:"accessKeyId"`
	AccessKeySecret string `json:"accessKeySecret"`
}

type AccessConfigForDeSEC struct {
	Token string `json:"token"`
}

type AccessConfigForDigitalOcean struct {
	AccessToken string `json:"accessToken"`
}

type AccessConfigForDingTalkBot struct {
	WebhookUrl string `json:"webhookUrl"`
	Secret     string `json:"secret"`
}

type AccessConfigForDiscordBot struct {
	BotToken         string `json:"botToken"`
	DefaultChannelId string `json:"defaultChannelId,omitempty"`
}

type AccessConfigForDNSLA struct {
	ApiId     string `json:"apiId"`
	ApiSecret string `json:"apiSecret"`
}

type AccessConfigForDogeCloud struct {
	AccessKey string `json:"accessKey"`
	SecretKey string `json:"secretKey"`
}

type AccessConfigForDuckDNS struct {
	Token string `json:"token"`
}

type AccessConfigForDynv6 struct {
	HttpToken string `json:"httpToken"`
}

type AccessConfigForEdgio struct {
	ClientId     string `json:"clientId"`
	ClientSecret string `json:"clientSecret"`
}

type AccessConfigForEmail struct {
	SmtpHost               string `json:"smtpHost"`
	SmtpPort               int32  `json:"smtpPort"`
	SmtpTls                bool   `json:"smtpTls"`
	Username               string `json:"username"`
	Password               string `json:"password"`
	DefaultSenderAddress   string `json:"defaultSenderAddress,omitempty"`
	DefaultReceiverAddress string `json:"defaultReceiverAddress,omitempty"`
}

type AccessConfigForFlexCDN struct {
	ServerUrl                string `json:"serverUrl"`
	ApiRole                  string `json:"apiRole"`
	AccessKeyId              string `json:"accessKeyId"`
	AccessKey                string `json:"accessKey"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForGcore struct {
	ApiToken string `json:"apiToken"`
}

type AccessConfigForGname struct {
	AppId  string `json:"appId"`
	AppKey string `json:"appKey"`
}

type AccessConfigForGoDaddy struct {
	ApiKey    string `json:"apiKey"`
	ApiSecret string `json:"apiSecret"`
}

type AccessConfigForGoEdge struct {
	ServerUrl                string `json:"serverUrl"`
	ApiRole                  string `json:"apiRole"`
	AccessKeyId              string `json:"accessKeyId"`
	AccessKey                string `json:"accessKey"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForGoogleTrustServices struct {
	EabKid     string `json:"eabKid"`
	EabHmacKey string `json:"eabHmacKey"`
}

type AccessConfigForHetzner struct {
	ApiToken string `json:"apiToken"`
}

type AccessConfigForHuaweiCloud struct {
	AccessKeyId     string `json:"accessKeyId"`
	SecretAccessKey string `json:"secretAccessKey"`
}

type AccessConfigForJDCloud struct {
	AccessKeyId     string `json:"accessKeyId"`
	AccessKeySecret string `json:"accessKeySecret"`
}

type AccessConfigForKubernetes struct {
	KubeConfig string `json:"kubeConfig,omitempty"`
}

type AccessConfigForLarkBot struct {
	WebhookUrl string `json:"webhookUrl"`
}

type AccessConfigForLeCDN struct {
	ServerUrl                string `json:"serverUrl"`
	ApiVersion               string `json:"apiVersion"`
	ApiRole                  string `json:"apiRole"`
	Username                 string `json:"username"`
	Password                 string `json:"password"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForMattermost struct {
	ServerUrl        string `json:"serverUrl"`
	Username         string `json:"username"`
	Password         string `json:"password"`
	DefaultChannelId string `json:"defaultChannelId,omitempty"`
}

type AccessConfigForNamecheap struct {
	Username string `json:"username"`
	ApiKey   string `json:"apiKey"`
}

type AccessConfigForNameDotCom struct {
	Username string `json:"username"`
	ApiToken string `json:"apiToken"`
}

type AccessConfigForNameSilo struct {
	ApiKey string `json:"apiKey"`
}

type AccessConfigForNetcup struct {
	CustomerNumber string `json:"customerNumber"`
	ApiKey         string `json:"apiKey"`
	ApiPassword    string `json:"apiPassword"`
}

type AccessConfigForNetlify struct {
	ApiToken string `json:"apiToken"`
}

type AccessConfigForNS1 struct {
	ApiKey string `json:"apiKey"`
}

type AccessConfigForPorkbun struct {
	ApiKey       string `json:"apiKey"`
	SecretApiKey string `json:"secretApiKey"`
}

type AccessConfigForPowerDNS struct {
	ServerUrl                string `json:"serverUrl"`
	ApiKey                   string `json:"apiKey"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForProxmoxVE struct {
	ServerUrl                string `json:"serverUrl"`
	ApiToken                 string `json:"apiToken"`
	ApiTokenSecret           string `json:"apiTokenSecret,omitempty"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForQiniu struct {
	AccessKey string `json:"accessKey"`
	SecretKey string `json:"secretKey"`
}

type AccessConfigForRainYun struct {
	ApiKey string `json:"apiKey"`
}

type AccessConfigForRatPanel struct {
	ServerUrl                string `json:"serverUrl"`
	AccessTokenId            int32  `json:"accessTokenId"`
	AccessToken              string `json:"accessToken"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForSafeLine struct {
	ServerUrl                string `json:"serverUrl"`
	ApiToken                 string `json:"apiToken"`
	AllowInsecureConnections bool   `json:"allowInsecureConnections,omitempty"`
}

type AccessConfigForSlackBot struct {
	BotToken         string `json:"botToken"`
	DefaultChannelId string `json:"defaultChannelId,omitempty"`
}

type AccessConfigForSSH struct {
	Host          string `json:"host"`
	Port          int32  `json:"port"`
	Username      string `json:"username"`
	Password      string `json:"password,omitempty"`
	Key           string `json:"key,omitempty"`
	KeyPassphrase string `json:"keyPassphrase,omitempty"`
	JumpServers   []struct {
		Host          string `json:"host"`
		Port          int32  `json:"port"`
		Username      string `json:"username"`
		Password      string `json:"password,omitempty"`
		Key           string `json:"key,omitempty"`
		KeyPassphrase string `json:"keyPassphrase,omitempty"`
	} `json:"jumpServers,omitempty"`
}

type AccessConfigForSSLCom struct {
	EabKid     string `json:"eabKid"`
	EabHmacKey string `json:"eabHmacKey"`
}

type AccessConfigForTelegramBot struct {
	BotToken      string `json:"botToken"`
	DefaultChatId int64  `json:"defaultChatId,omitempty"`
}

type AccessConfigForTencentCloud struct {
	SecretId  string `json:"secretId"`
	SecretKey string `json:"secretKey"`
}

type AccessConfigForUCloud struct {
	PrivateKey string `json:"privateKey"`
	PublicKey  string `json:"publicKey"`
	ProjectId  string `json:"projectId,omitempty"`
}

type AccessConfigForUniCloud struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type AccessConfigForUpyun struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type AccessConfigForVercel struct {
	ApiAccessToken string `json:"apiAccessToken"`
	TeamId         string `json:"teamId,omitempty"`
}

type AccessConfigForVolcEngine struct {
	AccessKeyId     string `json:"accessKeyId"`
	SecretAccessKey string `json:"secretAccessKey"`
}

type AccessConfigForWangsu struct {
	AccessKeyId     string `json:"accessKeyId"`
	AccessKeySecret string `json:"accessKeySecret"`
	ApiKey          string `json:"apiKey"`
}

type AccessConfigForWebhook struct {
	Url                        string `json:"url"`
	Method                     string `json:"method,omitempty"`
	HeadersString              string `json:"headers,omitempty"`
	AllowInsecureConnections   bool   `json:"allowInsecureConnections,omitempty"`
	DefaultDataForDeployment   string `json:"defaultDataForDeployment,omitempty"`
	DefaultDataForNotification string `json:"defaultDataForNotification,omitempty"`
}

type AccessConfigForWeComBot struct {
	WebhookUrl string `json:"webhookUrl"`
}

type AccessConfigForWestcn struct {
	Username    string `json:"username"`
	ApiPassword string `json:"password"`
}

type AccessConfigForZeroSSL struct {
	EabKid     string `json:"eabKid"`
	EabHmacKey string `json:"eabHmacKey"`
}
